### PR TITLE
Add stop() method to softglue device

### PR DESCRIPTION
**Pull request by prjemian**
... so user can interrupt a triggering sequence by calling its `.stop()` method.

### CODE REVIEW

**Review by MDecarabas — APPROVED** [review]

**Comment by prjemian** [comment]
Need to wait on a merge until:

- merge conflicts are resolved
- @qzhang234 has a chance to test (after SAS2024?)

**Comment by prjemian** [comment]
Setting this to draft to remind us...

**Comment by prjemian** [comment]
Also, refactor how softglue is triggered:

* so it can be triggered before the area detector
* so plan can proceed while softglue is triggering
* plan can stop triggering softglue when area detector is done

File: src/instrument/devices/softglue.py [hunk 1]
```diff
@@ -1,55 +1,50 @@
...
-            # has finished
-            self._done_acquiring()
-
-        # acq_signal.put(1, wait=False, callback=done_acquisition)  # ORIGINAL
-        trigger_value = self._sig_attrs[acq_signal.attr_name].trigger_value
-        acq_signal.put(trigger_value, wait=False, callback=done_acquisition)
-        return status
+    stop_pulses = Component(
+        EpicsSignal,
+        "8idi:softGlueA:OR-1_IN2_Signal",
```
Comment (MDecarabas) [inline src/instrument/devices/softglue.py:44]: Do we want to make the pv an argument so that we can reuse the same code later on for 9?

**Comment by sureshnaps** [comment]
Yes, these PVs are subject to change and would be good to not hard code it inside.

From: MDecarabas ***@***.***>
Date: Tuesday, November 5, 2024 at 10:40 AM
To: aps-8id-dys/bluesky ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [aps-8id-dys/bluesky] Add stop() method to softglue device (PR #102)
@ MDecarabas commented on this pull request. In src/instrument/devices/softglue. py: > - - self. subscribe(status. _finished, event_type=self. SUB_ACQ_DONE, run=False) - - def done_acquisition(**ignored_kwargs): - # Keyword arguments are ignored
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.

ZjQcmQRYFpfptBannerEnd

@MDecarabas commented on this pull request.

________________________________

In src/instrument/devices/softglue.py<https://urldefense.us/v3/__https:/github.com/aps-8id-dys/bluesky/pull/102*discussion_r1829676817__;Iw!!G_uCfscf7eWS!ZYSQ2qliVsZn_jsLPb-MlrjoG3TpsUgXLehIzwr46mSuiuW2pKBoF3iUqlbPAScNn_b46nTcY2WSILDKBYqibZXxRA$>:

> -

-        self.subscribe(status._finished, event_type=self.SUB_ACQ_DONE, run=False)

-

-        def done_acquisition(**ignored_kwargs):

-            # Keyword arguments are ignored here from the EpicsSignal

-            # subscription, as the important part is that the put completion

-            # has finished

-            self._done_acquiring()

-

-        # acq_signal.put(1, wait=False, callback=done_acquisition)  # ORIGINAL

-        trigger_value = self._sig_attrs[acq_signal.attr_name].trigger_value

-        acq_signal.put(trigger_value, wait=False, callback=done_acquisition)

-        return status

+    stop_pulses = Component(

+        EpicsSignal,

+        "8idi:softGlueA:OR-1_IN2_Signal",

Do we want to make the pv an argument so that we can reuse the same code later on for 9?

**Review by MDecarabas — APPROVED** [review]

**Pull request merged**